icu4c/source/common/serv.cpp

```diff
@@ -845,13 +833,13 @@ ICUService::createSimpleFactory(UObject* objToAdopt, const UnicodeString& id, UB
 }
 
 URegistryKey
-ICUService::registerFactory(ICUServiceFactory* factoryToAdopt, UErrorCode& status) 
+ICUService::registerFactory(ICUServiceFactory* factoryToAdopt, UErrorCode& status)
 {
     if (U_SUCCESS(status) && factoryToAdopt != NULL) {
         Mutex mutex(&lock);
 
         if (factories == NULL) {
-            factories = new UVector(deleteUObject, NULL, status);
+            factories = new UVector(uprv_deleteUObject, NULL, status);
             if (U_FAILURE(status)) {
```

Comment:
Might also want to check if `factories` is `nullptr` (in case of OOM), and set `status` to `U_MEMORY_ALLOCATION_ERROR` and return, so that it doesn't try to call `factories->insertElementAt` on nullptr below.

Reply:
Thanks for cleaning all this up!